### PR TITLE
fix index issue in patches modal  when removing patches

### PR DESCRIPTION
**Pull request by karolina-siemieniuk-morawska**
ezgif com-video-to-gif (1)

### CODE REVIEW

File: src/any-type-field/any-type-field.component.html [hunk 1]
```diff
@@ -1,12 +1,12 @@
 <div [ngSwitch]="schema.componentType">
   <div *ngSwitchCase="'table-list'">
-    <table-list-field [values]="value" [schema]="schema" [path]="path"></table-list-field>
+    <table-list-field [values]="value" [supportValues]="value" [schema]="schema" [path]="path"></table-list-field>
   </div>
   <div *ngSwitchCase="'complex-list'">
-    <complex-list-field [values]="value" [schema]="schema" [path]="path"></complex-list-field>
+    <complex-list-field [values]="value" [supportValues]="value" [schema]="schema" [path]="path"></complex-list-field>
```
Comment (drjova) [inline src/any-type-field/any-type-field.component.html:6]: what's the value o the value? Could you please give me an example?

Comment (karolina-siemieniuk-morawska) [reply src/any-type-field/any-type-field.component.html:6]: It's an Immutable List that populates tables, it's easiest to show on following screenshots (value of 'value' and corresponding table:
Screenshot 2023-08-09 at 14 17 48
Screenshot 2023-08-09 at 14 17 40

File: src/shared/services/json-store.service.ts [hunk 2]
```diff
@@ -249,4 +258,16 @@ export class JsonStoreService {
     const parentPathArray = pathArray.slice(0, -1);
     return this.pathUtilService.toPathString(parentPathArray);
   }
+
+  getPatchesIncludingPath(path: string): JsonPatch[] {
+    const pathsWithPatches = Object.keys(this.patchesByPath).filter(patch => patch.includes(path));
+    const patches = pathsWithPatches.map(p => this.patchesByPath[p]);
+    return [].concat(...patches);
```
[the remark this answers is not in the document]
Comment (karolina-siemieniuk-morawska) [reply src/shared/services/json-store.service.ts:265]: Those two solutions are not equal - concat method combined with spread operator results in flat array,  spread operator alone will still result in nested arrays, and we need flat (
Screenshot 2023-08-09 at 14 29 59
Screenshot 2023-08-09 at 14 29 47

[the remark this answers is not in the document]
Comment (drjova) [reply src/shared/services/json-store.service.ts:265]: 👍 we need flat

**Pull request merged**